src/__tests__/index.test.tsx

```diff
@@ -1 +1,149 @@
-it.todo('write a test');
+import WS from 'jest-websocket-mock';
+import { renderHook, act } from '@testing-library/react';
+import { JellyfishContextProvider, useJellyfishClient } from '..';
+import { PeerMessage } from '../protos/jellyfish/peer_notifications';
+import { NativeModules } from 'react-native';
+
+let sendEvent: null | ((mediaEvent: string) => void) = null;
+
+jest.mock('@jellyfish-dev/react-native-membrane-webrtc');
+jest.mock('react-native', () => ({
+  NativeModules: {
+    Membrane: {
+      receiveMediaEvent: jest.fn(),
+      create: jest.fn(),
+    },
+  },
+  Platform: {
+    select: jest.fn(),
+  },
+  NativeEventEmitter: jest.fn().mockImplementation(() => ({
+    addListener: jest.fn((_, _sendEvent) => {
+      sendEvent = _sendEvent;
+      return { remove: jest.fn() };
+    }),
+  })),
+  UIManager: {
+    getViewManagerConfig: jest.fn(),
+  },
+}));
+
+function encodePeerMessage(peerMessage: object) {
+  const encodedAsNodeBuffer = PeerMessage.encode(peerMessage).finish();
+
+  const newBuffer = new ArrayBuffer(encodedAsNodeBuffer.byteLength);
+  const newBufferView = new Uint8Array(newBuffer);
+  newBufferView.set(encodedAsNodeBuffer, 0);
+
+  return newBuffer;
+}
+
+describe('JellyfishClient', () => {
+  afterEach(() => {
+    WS.clean();
+  });
+
+  const setUpAndConnect = async () => {
+    const server = new WS('ws://localhost:1234');
```

Comment:
I'd extract the address to a variable sice it is used quite a few times in this test file.